modules/wallet/transactionbuilder.go

```diff
@@ -9,6 +9,90 @@ import (
 	"github.com/NebulousLabs/Sia/types"
 )
 
+// walletSpender allows clients to retrieve and modify keys and outputs in a
+// wallet in order to spend siacoins.
+type walletSpender interface {
+	// acquireLock acquires a mutex on walletSpender. Clients must call
+	// acquireLock before calling any other function in walletSpender and must
+	// call releaseLock before exiting scope.
+	acquireLock()
+	releaseLock()
```

Comment:
These don't belong here. Check out the mutex section:

https://github.com/NebulousLabs/Sia/blob/master/doc/Developers.md

Sia is a pretty complex project, and the locking is already really complex in its simplified form. As soon as you start letting modules grab and release locks manually, you open a whole set of complexity and places for things like deadlocks and race conditions to hide. It gets out of control quickly. The rule that we use to keep things as simple as possible is that modules need to handle their own locking internally. If they expose a function through an interface, then that function needs to be thread-safe.

There's another rule that's only just being introduced. You haven't violated it, but I thought I'd mention it anyway since it's not documented anywhere yet. A module that is holding a write-lock on itself is not allowed to make any external calls. This prevents a lot of potential deadlocks and eliminates the need to understand how the locking structure of foreign interfaces operates.